File: solutions/java/export.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class export {

    public static void main(String[] args) {

		try {
			if(args.length != 1) {
				System.out.println("syntax: export \"search query\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);
	        InputStream response = service.export(args[0]);

	        int chr = 0;
			while (chr != -1) System.out.write(chr = response.read());
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: solutions/python/export.py

import splunklib.client as client
import utils
import sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 2:
    print "Syntax: export.py \"search query\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
    response = service.jobs.export(sys.argv[1])                
    print response

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
